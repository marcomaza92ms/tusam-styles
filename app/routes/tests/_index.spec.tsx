import { createRoutesStub } from "react-router";
import Index from "~/routes/_index";
import { render } from "@testing-library/react";

const setup = (initialEntries: string) => {
  const App = createRoutesStub([
    {
      path: "*",
      Component: Index,
    },
  ]);

  return render(<App initialEntries={[initialEntries]} />);
};

describe("GIVEN a root example page", () => {
  it("SHOULD display successfully", () => {
    const wrapper = setup("/");

    expect(wrapper.queryByText(/welcome to remix/i)).toBeDefined();

    expect(wrapper.asFragment()).toMatchSnapshot();
  });
});
