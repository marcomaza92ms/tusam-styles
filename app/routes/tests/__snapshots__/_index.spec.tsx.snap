// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`GIVEN a root example page > SHOULD display successfully 1`] = `
<DocumentFragment>
  <header>
    <nav
      class="w-full flex flex-row justify-center items-center my-4"
    >
      <ul
        class="flex flex-row gap-4"
      >
        <li>
          <a
            href="/"
          >
            Homepage
          </a>
        </li>
        <li>
          <a
            href="/css-props"
          >
            CSS Properties
          </a>
        </li>
        <li>
          <a
            href="/typed-om"
          >
            Typed OM
          </a>
        </li>
        <li>
          <a
            href="/paint-api"
          >
            Paint API
          </a>
        </li>
        <li>
          <a
            href="/layout-api"
          >
            Layout API
          </a>
        </li>
      </ul>
    </nav>
  </header>
  <main
    class="content"
  >
    <h1>
      Tusam Styles
    </h1>
    <h2>
      Behold!, for the magic you are about to see
    </h2>
  </main>
  <footer
    class="flex flex-row items-center justify-center my-8"
  >
    Â© 2025 Tusam Styles
  </footer>
</DocumentFragment>
`;
